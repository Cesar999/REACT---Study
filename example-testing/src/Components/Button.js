import React, {useState} from 'react';

function Button(props){
    const [value, setValue] = useState('');
    const handleClick = () => {
        setValue('');
    };
    return (
        <>
            <label htmlFor={`button-${props.label}`}>{props.label}</label>
            <button data-testid="button-component" id={`button-${props.label}`} onClick={handleClick}>{props.label}</button>
        </>
    )
}

export default Button;