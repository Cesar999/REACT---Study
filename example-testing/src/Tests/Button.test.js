import { render } from '@testing-library/react';
import Button from '../Components/Button';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer
      .create(<Button label="Click Me"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders Button Component', () => {
    const wrapper = render(<Button label="Click Me"/>);
    const {getByTestId} = wrapper;
    expect(getByTestId('button-component').textContent).toBe("Click Me")
});

