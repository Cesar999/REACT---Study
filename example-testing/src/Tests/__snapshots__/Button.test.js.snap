// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <label
    htmlFor="button-Click Me"
  >
    Click Me
  </label>,
  <button
    data-testid="button-component"
    id="button-Click Me"
    onClick={[Function]}
  >
    Click Me
  </button>,
]
`;
